### PR TITLE
Modify logic to not throw error for singleton coordinates (with no bounds)

### LINKED ISSUE #304: [Bug]: ValueError: Cannot generate bounds for a coordinate of length <= 1. Using decode_times optional arg
### What happened?

Attempting to open a dataset with `time:units = "months since 1955-01-01 00:00:00" ;` has problems, adding `decode_times=False` gets around the pandas issue (#282), but this leads to a `xcdat/bounds.py` issue instead: `ValueError: Cannot generate bounds for a coordinate of length <= 1.`

### What did you expect to happen?

xcdat could handle the poorly formed file and open the dataset.

### Minimal Complete Verifiable Example

```python
In [2]: from xcdat import open_dataset

In [3]: f = "~/Shared/obs_data/WOD18/190312/woa18_decav_t00_04.nc"

In [4]: wH = open_dataset(f)

Or 

In [5]: wH = open_dataset(f, decode_times=False)
```

### Relevant log output

```python
Traceback (most recent call last):

  File "/tmp/ipykernel_177169/1773024523.py", line 1, in <cell line: 1>
    wH = open_dataset(f)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/dataset.py", line 89, in open_dataset
    ds = decode_non_cf_time(ds)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/dataset.py", line 320, in decode_non_cf_time
    data = [ref_date + pd.DateOffset(**{units: offset}) for offset in time.data]

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/dataset.py", line 320, in <listcomp>
    data = [ref_date + pd.DateOffset(**{units: offset}) for offset in time.data]

  File "pandas/_libs/tslibs/offsets.pyx", line 444, in pandas._libs.tslibs.offsets.BaseOffset.__add__

  File "pandas/_libs/tslibs/offsets.pyx", line 450, in pandas._libs.tslibs.offsets.BaseOffset.__add__

  File "pandas/_libs/tslibs/offsets.pyx", line 180, in pandas._libs.tslibs.offsets.apply_wraps.wrapper

  File "pandas/_libs/tslibs/offsets.pyx", line 1092, in pandas._libs.tslibs.offsets.RelativeDeltaOffset._apply

  File "pandas/_libs/tslibs/timestamps.pyx", line 1399, in pandas._libs.tslibs.timestamps.Timestamp.__new__

  File "pandas/_libs/tslibs/conversion.pyx", line 436, in pandas._libs.tslibs.conversion.convert_to_tsobject

  File "pandas/_libs/tslibs/conversion.pyx", line 517, in pandas._libs.tslibs.conversion.convert_datetime_to_tsobject

  File "pandas/_libs/tslibs/np_datetime.pyx", line 120, in pandas._libs.tslibs.np_datetime.check_dts_bounds

OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 2315-07-01 00:00:00


In [5]: wH = open_dataset(f, decode_times=False)
Traceback (most recent call last):

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 178, in get_bounds
    bounds_key = coord_var.attrs["bounds"]

KeyError: 'bounds'


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 145, in add_missing_bounds
    self.get_bounds(axis)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 180, in get_bounds
    raise KeyError(

KeyError: "The coordinate variable 'time' has no 'bounds' attr. Set the 'bounds' attr to the name of the bounds data variable."


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 178, in get_bounds
    bounds_key = coord_var.attrs["bounds"]

KeyError: 'bounds'


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 220, in add_bounds
    self.get_bounds(axis)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 180, in get_bounds
    raise KeyError(

KeyError: "The coordinate variable 'time' has no 'bounds' attr. Set the 'bounds' attr to the name of the bounds data variable."


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/tmp/ipykernel_177169/170555424.py", line 1, in <cell line: 1>
    wH = open_dataset(f, decode_times=False)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/dataset.py", line 95, in open_dataset
    ds = _postprocess_dataset(ds, data_var, center_times, add_bounds, lon_orient)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/dataset.py", line 482, in _postprocess_dataset
    dataset = dataset.bounds.add_missing_bounds()

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 147, in add_missing_bounds
    self._dataset = self.add_bounds(axis, width)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 225, in add_bounds
    dataset = self._add_bounds(axis, width)

  File "~/mambaforge/envs/xcdat030spy532/lib/python3.10/site-packages/xcdat/bounds.py", line 271, in _add_bounds
    raise ValueError("Cannot generate bounds for a coordinate of length <= 1.")

ValueError: Cannot generate bounds for a coordinate of length <= 1.
```

### Anything else we need to know?

Adding the additional argument
```python
wH = open_dataset(f, decode_times=False, add_bounds=False)
```
opens the file fine

### Environment

0.3.0 - conda-forge build

In [12]: xr.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:04:59) [GCC 10.3.0]
python-bits: 64
OS: Linux
OS-release: 3.10.0-1160.62.1.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: ('en_US', 'UTF-8')
libhdf5: 1.12.2
libnetcdf: 4.8.1

xarray: 2022.6.0
pandas: 1.4.3
numpy: 1.22.4
scipy: 1.9.0
netCDF4: 1.6.0
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.1
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2022.8.0
distributed: 2022.8.0
matplotlib: None
cartopy: None
seaborn: None
numbagg: None
fsspec: 2022.7.1
cupy: None
pint: None
sparse: 0.13.0
flox: None
numpy_groupies: None
setuptools: 63.4.2
pip: 22.2.2
conda: None
pytest: None
IPython: 7.33.0
sphinx: 5.1.1
~/mambaforge/envs/xcdat031spy532/lib/python3.10/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

### PR DESCRIPTION
**Pull request by pochedls**
Handle axis with single timesteps when adding bounds

## Description
When a coordinate has a length of 0 or 1, do not throw an error (in order to accommodate single timestep datasets). Simply do not add bounds and warn the user. 

- Closes #304

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass with my changes (locally and CI/CD build)
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: tests/test_bounds.py [hunk 1]
```diff
@@ -166,9 +166,6 @@ def test_raises_errors_for_data_dim_and_length(self):
         # If coords dimensions does not equal 1.
         with pytest.raises(ValueError):
             ds.bounds.add_bounds("Y")
-        # If coords are length of <=1.
-        with pytest.raises(ValueError):
-            ds.bounds.add_bounds("X")
```
Comment (pochedls) [inline tests/test_bounds.py:171]: Do we need to test for this case: 

when a dimension with length <= 1 that does not have bounds, it does not get bounds added to the axis and a warning is thrown

Comment (pochedls) [reply tests/test_bounds.py:171]: I added a test to check that the original dataset is returned (no bounds added) and suppressed the warning.

File: xcdat/bounds.py [hunk 2]
```diff
@@ -117,8 +117,16 @@ def keys(self) -> List[str]:
...
+        This function loops through the Dataset's axes and attempts to adds
+        bounds to its coordinates if they don't exist. The coordinates must meet
+        the following criteria in order to add bounds:
+
+          1. The axis for the coordinates are "X", "Y", "T", or "Z"
+          2. Coordinates are a single dimension, not multidimensional
+          3. Coordinates are a length > 1 (not singleton)
+          4. Bounds must not already exist.
+             * Determined by attempting to map the coordinate variable's
+             "bounds" attr (if set) to the bounds data variable of the same key.
```
Comment (tomvothecoder) [inline xcdat/bounds.py:129]: Added explicit criteria on when bounds can be added for coordinates.

File: xcdat/bounds.py [hunk 3]
```diff
@@ -133,24 +141,30 @@ def add_missing_bounds(self, width: float = 0.5) -> xr.Dataset:
...
                 pass
 
-            # determine if the axis is also a dimension by determining
-            # if there is overlap between the CF axis names and the dimension
-            # names. If not, skip over axis for validation.
-            if len(set(CF_NAME_MAP[axis]) & set(self._dataset.dims.keys())) == 0:
+            try:
+                self._dataset = self.add_bounds(axis, width)
+            except ValueError:
                 continue
```
Comment (tomvothecoder) [inline xcdat/bounds.py:166]: Try to add bounds or continue (if coords are multidimensional or singleton).

File: tests/test_bounds.py [hunk 4]
```diff
@@ -60,6 +60,7 @@ def setup(self):
     def test_adds_bounds_to_the_dataset(self):
         ds = self.ds_with_bnds.copy()
 
+        # Delete the lat and lon bounds and remove the "bounds" attr mapping
```
Comment (tomvothecoder) [inline tests/test_bounds.py:63]: TODO: Delete this outdated comment.

**Review by tomvothecoder — COMMENTED** [review]
Hey @pochedls, I did a little bit of refactoring to handle the case of attempting to add bounds for multidimension/singleton coords.

Let me know what you think. If it checks out, please merge.

Note, some review comments might say "Outdated" because I updated some comments in subsequent commits. You can ignore these warnings.

File: tests/test_bounds.py (hunk 4 above)
Comment (tomvothecoder) [inline tests/test_bounds.py:63]:
```suggestion
```

File: tests/test_bounds.py [hunk 5]
```diff
@@ -148,7 +167,7 @@ def test_raises_error_if_bounds_already_exist(self):
         with pytest.raises(ValueError):
             ds.bounds.add_bounds("Y")
 
-    def test_raises_errors_for_data_dim_and_length(self):
+    def test_raises_errors_for_data_dim_and_length(self, caplog):
```
Comment (tomvothecoder) [inline tests/test_bounds.py:170]:
```suggestion
    def test_raises_errors_for_data_dim_and_length(self):
```

File: xcdat/bounds.py [hunk 6]
```diff
@@ -133,24 +141,30 @@ def add_missing_bounds(self, width: float = 0.5) -> xr.Dataset:
...
+            # is overlap between the CF axis names and the dimension names. If
+            # not, skip over axis for validation.
+            if len(set(CF_NAME_MAP[axis]) & set(self._dataset.dims.keys())) == 0:
+                continue
+
+            # Check if bounds also exist using the "bounds" attribute.
+            # Otherwise, try to add bounds if it meets the function's criteria.
             try:
-                coord_var = get_axis_coord(self._dataset, axis)
+                self.get_bounds(axis)
```
Comment (tomvothecoder) [inline xcdat/bounds.py:159]:
```suggestion
                self.get_bounds(axis)
                continue
```

I forgot to add `continue` if bounds are found.

File: xcdat/bounds.py [hunk 7]
```diff
@@ -133,24 +141,31 @@ def add_missing_bounds(self, width: float = 0.5) -> xr.Dataset:
...
+                continue
```
Comment (tomvothecoder) [inline xcdat/bounds.py:160]: In an earlier commit, I forgot to add a `continue` if bounds are found.

File: xcdat/bounds.py [hunk 8]
```diff
@@ -272,9 +295,15 @@ def _add_bounds(self, axis: CFAxisName, width: float = 0.5) -> xr.Dataset:
 
         # Validate coordinate shape and dimensions
         if coord_var.ndim != 1:
-            raise ValueError("Cannot generate bounds for multidimensional coordinates.")
+            raise ValueError(
+                "Cannot generate bounds for coordinate variable '{coord_var.name}"
```
Comment (tomvothecoder) [inline xcdat/bounds.py:299]:
```suggestion
                f"Cannot generate bounds for coordinate variable '{coord_var.name}"
```

File: xcdat/bounds.py [hunk 9]
```diff
@@ -272,9 +295,15 @@ def _add_bounds(self, axis: CFAxisName, width: float = 0.5) -> xr.Dataset:
 
         # Validate coordinate shape and dimensions
         if coord_var.ndim != 1:
-            raise ValueError("Cannot generate bounds for multidimensional coordinates.")
+            raise ValueError(
+                f"Cannot generate bounds for coordinate variable '{coord_var.name}"
+                "because it is multidimensional coordinates."
```
Comment (tomvothecoder) [inline xcdat/bounds.py:300]:
```suggestion
                " because it is multidimensional coordinates."
```

File: xcdat/bounds.py [hunk 10]
```diff
@@ -272,9 +295,15 @@ def _add_bounds(self, axis: CFAxisName, width: float = 0.5) -> xr.Dataset:
 
         # Validate coordinate shape and dimensions
         if coord_var.ndim != 1:
-            raise ValueError("Cannot generate bounds for multidimensional coordinates.")
+            raise ValueError(
+                f"Cannot generate bounds for coordinate variable '{coord_var.name}"
```
Comment (tomvothecoder) [inline xcdat/bounds.py:299]:
```suggestion
                f"Cannot generate bounds for coordinate variable '{coord_var.name}'"
```

**Pull request merged**